tssc/step_implementers/push_artifacts/maven.py

```diff
@@ -137,27 +146,35 @@ def required_runtime_step_config_keys():
         """
         return REQUIRED_CONFIG_KEYS
 
-    def _validate_runtime_step_config(self, runtime_step_config):
-        """
-        Validates the given `runtime_step_config` against the required step configuration keys.
-
-        Parameters
-        ----------
-        runtime_step_config : dict
-            Step configuration to use when the StepImplementer runs the step with all of the
-            various static, runtime, defaults, and environment configuration munged together.
-
-        Raises
-        ------
-        AssertionError
-            If the given `runtime_step_config` is not valid with a message as to why.
-        """
-        super()._validate_runtime_step_config(runtime_step_config) #pylint: disable=protected-access
+    def _get_version(self):
+        # Required: Get the generate-metadata.version
+        if (self.get_step_results('generate-metadata') and
+                self.get_step_results('generate-metadata').get('version')):
+            return self.get_step_results('generate-metadata')['version']
+        raise ValueError('generate-metadata results missing version')
 
-        assert ( \
-            all(element in runtime_step_config for element in AUTHENTICATION_CONFIG) or \
-            not any(element in runtime_step_config for element in AUTHENTICATION_CONFIG) \
-        ), 'Either username or password is not set. Neither or both must be set.'
+    def _get_artifacts(self):
+        # Required: Get the package.artifacts
+        if (self.get_step_results('package') and
+                self.get_step_results('package').get('artifacts')):
+            return self.get_step_results('package')['artifacts']
+        raise ValueError('package results missing artifacts')
+
+    def _settings_file(self):
```

Comment:
@ppremru @akim309 suggest calling this `__generate_maven_settings` for consistancy.